floss/language/go/extract.py

```diff
@@ -0,0 +1,388 @@
+# Copyright (C) 2023 Mandiant, Inc. All Rights Reserved.
+
+import re
+import sys
+import struct
+import logging
+import pathlib
+import argparse
+from typing import List, Iterable, Optional
+from itertools import chain
+
+import pefile
+
+from floss.results import StaticString, StringEncoding
+
+logger = logging.getLogger(__name__)
+
+MIN_STR_LEN = 6
+
+
+def yield_string(binary_string, addr, min_length):
```

Comment:
- rename to `decode_and_validate` or similar to better indicate the functionality
- please add type hints